tests/unit/test_utils.py

```diff
@@ -409,3 +410,25 @@ def test_can_escape_recursive_shapes(self):
         }
         self.walker.walk(self.get_shape_model('Recursive'), self.visitor)
         self.assert_visited_shapes(['Recursive'])
+
+
+@pytest.fixture(scope='module')
```

Comment:
Let's remove the `scope` argument for this fixture. Specifically, I don't think we want the `scope` to be `module` because it will mean this instance will only be instantiated once, which means a test case can change the state of the argument model and have it affect the state in a subsequent test case. By removing the `scope`, it will make sure the fixture is instantiated brand new for each test case and thus have a clean state.